### ISSUE TITLE
Setting withHistory disables historyInputFile and historyOutputFile

### ISSUE THREAD

**Issue by peter-janssen** [hcoles/pitest#293]
It took me a while to figure this out and I had to go through your code to find it. In MojoToReportOptionsConverter the following code uses the default temp location when withHistory is set to true. In that case the historyInputFile and historyOutputFile are ignored. To me it would be more logical that when withHistory is set a check is done to see if the locations are set and use those instead of the tmp location. If you want I can make a pull request but I'd have to get set up first. 

```
private void determineHistory(final ReportOptions data) {
    if (this.mojo.useHistory()) {
      useHistoryFileInTempDir(data);
    } else {
      data.setHistoryInputLocation(this.mojo.getHistoryInputFile());
      data.setHistoryOutputLocation(this.mojo.getHistoryOutputFile());
    }
}
```

**Comment by hcoles** [comment]
withHistory is intended as a shortcut for setting the values of input and output to the same location.

It doesn't really make sense set both withHistory and specific locations. Perhaps the best behaviour would be to throw an error if both are set?

**Comment by peter-janssen** [comment]
Throwing an error or warning works aswell. Too me setting both withHistory and specific locations does make sense though. And ignoring the specific locations when withHistory is set is counter-intuative to me. But maybe that is just me ;-)
I do see a use case where you want to set both. Let's say you use a custom location for your history. But I want to do a manual run without history (for debugging purposes or something). I'd set -DwithHistory=false. But the location is set in my pom file, the result being that it still records history for that run.

**Comment by bentlor** [comment]
I agree with peter-janssen. It is essential that these are checked in the correct order, and both have effects as described in the FAQ. I was struggling to set up the suite because I thought that they have to be used together in order to use a different path *and* use history. Also, the log doesn't display "Using history file ..." when only using the location parameters, which suggests that there is no history file being used.
TL;DR: At least document in the FAQ how these 3 properties actually work.

**Comment by ilgun** [comment]
Agreeing with @peter-janssen  as well, it confused me too :)

**Comment by evgygor** [comment]
Really confusing,  the issue should  be noticed in the readme
